crates/sui-node/src/lib.rs

```diff
@@ -172,19 +184,19 @@ impl SuiNode {
 
         let (certified_checkpoint_output, forward_to_state_sync_task) =
             sui_core::checkpoints::SendCheckpointToStateSync::new();
-        let checkpoint_store = CheckpointStore::new(&config.db_path().join("checkpoints"));
+
+        // init the task that forwards checkpoints to state_sync
+        forward_to_state_sync_task.start(state_sync_handle.clone());
```

Comment:
After this pr we could remove this abstraction. But I can send a pr for that as a follow up